packages/opentelemetry-sdk-trace-web/src/utils.ts

```diff
@@ -296,7 +296,7 @@ export function parseUrl(url: string): HTMLAnchorElement {
  * simplified to contain id
  */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function getElementXPath(target: any, optimised?: boolean) {
+export function getElementXPath(target: any, optimised?: boolean): string {
```

Comment:
the parameter `target: any` still has a lint warning, but I couldn't find the proper type for it. If you think it should be `any`, I will disable the lint warning for this line.

Reply:
Looks like the type is `EventTarget` which is defined in [`lib.dom.d.ts`](https://github.com/microsoft/TypeScript/blob/b12d5c5dc5106a9ee040cbf4acc32f80a8032c7f/lib/lib.dom.d.ts#L5165) . The only use I could find of it in either of our repos is here: [`@opentelemetry/instrumentation-user-interaction/src/instrumentation.ts#L101`](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/7a7d3a35e66678cf7d11fb9c89b8a26f2d4bfd6b/plugins/web/opentelemetry-instrumentation-user-interaction/src/instrumentation.ts)

Personally, it seems to me like this function should be in the user interaction plugin and not here at all, but unfortunately we already released this as 1.0 so it can't be removed. You can decide if you want to keep `any` here or use the more specific `EventTarget` type.

We can add a `@deprecated` notice to the tsdoc and move the function to the plugin maybe? @vmarchaud wdyt

Reply:
I tried it already. Using `EventTarget` reveals other errors which I couldn't fix.

Reply:
`any` is fine then. I think the plugin uses the zone.js modified version of EventTarget. Even more reason to deprecate this method IMO

Reply:
> We can add a @deprecated notice to the tsdoc and move the function to the plugin maybe? @vmarchaud wdyt

I think we can both add a deprecated tag and a warning at the top of the jsdoc comment

Reply:
If you agree, I will create another PR later to add the deprecated warning and add a duplicate of this function to the plugin (actually I don't know which plugin).
I imagine having the following deprecation warning message:
> This function will be removed in v2. Consider using the `getElementXPath` function in ... plugin instead.

Reply:
`v1` needs to be supported for around at least 3 years so I would really rethink adding deprecation already on start to be able to see this warning for the next 3 years ...

Reply:
I believe it is likely that nobody even knows this function is here or what it does. Only one plugin uses it and that plugin and this function were both written by the same person. I would prefer to remove it entirely, but that's obviously not possible so deprecation is the best we can do. I honestly don't mind having a deprecation warning around for multiple years. Many projects including node have done that.

Reply:
For now we'll leave the deprecation warning off since we haven't moved the function to the plugin yet. @alisabzevari `any` is fine for now.

Reply:
Got it. The change to `any` is already pushed.